src/sensor/pingsensor.cpp

```diff
@@ -39,7 +39,9 @@ void PingSensor::writeMessage(const ping_message& msg) const
 
 void PingSensor::handleMessagePrivate(const ping_message& msg)
 {
+#ifndef QT_NO_DEBUG
```

Comment:
I don't think we should repurpose this flag, it will loose asserts and other debugging things.

Try using QT_NO_DEBUG_OUTPUT, then we do not have to guard all of our qdebug statements.

Or make a new flag PING360_OPTIMIZE to guard here.

Reply:
We are just avoiding verbose messages in release that could impact performance.

Reply:
> We are just avoiding verbose messages in release that could impact performance.

That is what you mean to do, but you are also disabling any asserts() and so on that appear elsewhere in the program. The QT_NO_DEBUG flag is not designed for this purpose. You should have a PING360_OPTIMIZE flag that is independent of QT_NO_DEBUG.

Reply:
I see you are trying to target release builds only with this, where I guess the QT_NO_DEBUG flag is already set.

These debug messages are to help us find problems with users. We don't have debug build for mac and linux though. Also, I don't think it is a good trade to disable this for release builds.

Reply:
QT_NO_DEBUG is the "release" definition, there is no `QT_RELEASE` if that is what you are thinking. 


> The QT_NO_DEBUG flag is not designed for this purpose.
Can you point where is the purpose of such flag ? AFIK this flag is the "no" version of QT_DEBUG that is used to detect code compiled in debug mode. 

@tcanabrava any idea ?

Reply:
I see that this only covers release builds now @patrickelectric that is ok with me to use the same flag for that. Thank you for explaining it.

BUT I don't want to take these messages out of release builds. I approve this patch except for this commit, it should be removed.

Reply:
This messages are a bit annoying and also time consuming, removing this commit will result in almost 4% performance time, making the execution time from 7.7s down to 7.3s, with the others commits this brings down to 7s and with some that are still in development under 7s in execution time. This messages are not removed, if debug is necessary, you can use the debug builds, that's why we have it.

Reply:
The issue is that even with the debug output removed with Q_NO_DEBUG_OUTPUT, the calculations done within it would still be done. I don't really like the #ifdef as it's hard to track the usecase when people compile it's own code, but we could do something in the lines of

if (debugCategoryIsEnabled) {
   qDebug() 
}

then it would be only a jmp and quite fast.

Reply:
With that such message will not saved in the log system.

Reply:
We cannot remove these messages at this point in time. I won't accept it. Perhaps later when everything is stable.

>  4% performance time lost, making the execution time from 7.7s down to 7.3s

This is not a good trade in my opinion. There are better changes we can make to improve scan speed drastically, without sacrificing our valuable logging debug information.

> you can use the debug builds, that's why we have it.

Our users cannot use debug builds, because we do not have it for mac and linux.

Reply:
We can do as Tomaz said, the messages will be there.

Reply:
I'll remove this commit for now to help this PR before next text release